### ISSUE TITLE
3.1.4 memory leak

### ISSUE THREAD

**Issue by rusu24edward** [visit-dav/live-customer-response#236]
We're trying to [combine `visit-users` email list with GitHub issues](https://github.com/visit-dav/live-customer-response/wiki/How-the-new-GitHub-visit-users-Email-Integration-Works).
When replying on `visit-users`, please reply **only** to emails with a *Subject*
line that includes `[visit-dav/live-customer-response]`.

---
Visit team,
 
My team regularly uses Visit for visualizing large (>20GB) .hdf5 files containing volume data from our CFD simulations; Visit 3.1.0 and 3.1.4 handled visualizing this data just fine until our CFD code was refactored to utilize a box size of (8^3) when generating a Cartesian mesh; attempting to visualize.hdf5 data with many boxes of this size always crashes Visit (3.1.0 and 3.1.4); at least, it crashes the parallel job we run visit on, specifically our compute nodes run out of memory. Other common box sizes (16^3, 32^3) appear to not create this problem so far. We were wondering if the development team has any fixes or awareness of this issue and how to remedy it going forward.
 
Thanks for your help!
-Scott

**Comment by rusu24edward** [comment]
Hi Scott,

Do you have [scalable rendering](https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/gui_manual/Preferences/Rendering_Options_Window.html?highlight=scalable%20rendering#scalable-rendering) activated?

Eddie

**Comment by scottneuhoff** [comment]
Eddie,

Yes I've got it set to the default, which uses scalable rendering whenever 200K polys is exceeded (which is effectively always for my purposes.)

**Comment by rusu24edward** [comment]
Hi @scottneuhoff,

Can you run in debug mode and send me the [vlog files](https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/gui_manual/Preferences/File_Locations.html?highlight=debug#visit-debug-log-vlog-files)? You can zip them up and add them to this issue.

Eddie

**Comment by scottneuhoff** [comment]
Eddie,

I have those vlog files however I'm wary of attaching them here for security concerns, am I able to send them to you privately?

**Comment by rusu24edward** [comment]
Hi @scottneuhoff,

Yes, you can send them to my email. rusu1@llnl.gov

**Comment by rusu24edward** [comment]
Note from Scott:
Just wanted to give you a quick update; my group discovered that Visit was crashing when loading these volume solution HDF5 files because of ghost cell calculations. We found that if we output a volume file that does not contain solution values in the ghost cells and rely on Visit to compute those values, Visit blows out the memory trying to compute them. However, outputting volume files with the ghost cell values already computed allowed Visit to load and visualize them successfully. Perhaps there is something deeper going on with how Visit computes its ghost cell values; we aren’t sure; but for now we will be outputting those values. Ideally it would save us some file space if we did not have to, but it is something that can be explored in the future.
